packages/user-operation-controller/src/UserOperationController.ts

```diff
@@ -517,4 +660,143 @@ export class UserOperationController extends BaseController<
       },
     );
   }
+
+  async #requestApproval(metadata: UserOperationMetadata): Promise<AddResult> {
+    const { id, origin } = metadata;
+    const type = ApprovalType.Transaction;
+    const requestData = { txId: id };
+
+    return (await this.messagingSystem.call(
+      'ApprovalController:addRequest',
+      {
+        id,
+        origin,
+        type,
+        requestData,
+        expectsResult: true,
+      },
+      true, // Should display approval request to user
+    )) as Promise<AddResult>;
```

Comment:
Despite the type information seeming all set up[^1], the type isn't being inferred as I expected. Have you come across this before or know why it might be happening?

[^1]: https://github.com/MetaMask/core/blob/83d37dcf390342a8ec6fb0c3a9886267a890e1cb/packages/approval-controller/src/ApprovalController.ts#L265-L271

Reply:
I assume this is a side effect of having overloads for that method in the `ApprovalController` since sometimes it returns `Promise<AddResult` and sometimes `Promise<unknown`.

Shouldn't matter now though since I'm casting to provide a typed `value` as you suggested.